tox.ini

```diff
@@ -1,42 +1,34 @@
 [tox]
 toxworkdir={env:TOXWORKDIR:{toxinidir}/.tox}
 envlist =
-    {py35,py36,py37}-django20
-    {py35,py36,py37}-django21
-    {py35,py36,py37,py38,py39}-django22
-    {py36,py37,py38,py39}-django30
-    {py36,py37,py38,py39}-django31
-    {py36,py37,py38,py39,py310}-django32
+    {py38,py39,py310}-django32
     {py38,py39,py310}-django40
-    {py38,py39,py310}-django41
-    {py38,py39,py310}-django42
+    {py38,py39,py310,py311}-django41
+    {py38,py39,py310,py311}-django42
+    {py310,py311,py312}-django42
     flake8
 skip_missing_interpreters = True
 
 [flake8]
 max-line-length = 120
 
 [testenv]
-commands = python ./manage.py test
+commands = python -W error::DeprecationWarning -W error::PendingDeprecationWarning ./manage.py test
```

Comment:
This pattern helps prevent deprecation warnings from affecting users. I use these flags in my open source projects.